pandas/tests/io/excel/test_readers.py

```diff
@@ -130,8 +132,15 @@ def df_ref(datapath):
     return df_ref
 
 
-def adjust_expected(expected: DataFrame, read_ext: str) -> None:
+def get_exp_unit(read_ext: str, engine: str | None) -> str:
```

Comment:
So your ultimate vision here is different engines will return different bases? Or is there just considered a temporary solution?

Reply:
What makes ods files or the calamine reader different here? My first thought is that is surprising behavior to have those be the only ones return seconds

Reply:
no idea.  id be OK with coercing them all to micros

Reply:
December 31, 9999 23:59:59.999999 is also the highest value supported by the stdlib datetime.  microseconds _can_ go further than that, but most of the time when we get microseconds it is because we got a stdlib datetime objects

Reply:
(btw i definitely want to have this discussion but would prefer it to happen in #55901, this PR is just aimed at diff-trimming to make that one easier)